tests/core/test_topics.py

```diff
@@ -404,33 +414,39 @@ async def test_remove_user_topics_on_user_delete(client, app):
         ],
     )
 
-    app.data.insert(
+    await create_entries_for(
         "user_topic_folders",
         [
-            {"name": "delete", "user": PUBLIC_USER_ID},
-            {"name": "skip", "user": TEST_USER_ID},
+            {"_id": ObjectId(), "name": "delete", "user": PUBLIC_USER_ID, "section": "wire"},
+            {"_id": ObjectId(), "name": "skip", "user": TEST_USER_ID, "section": "wire"},
         ],
     )
 
-    topics, _ = app.data.find("topics", req=None, lookup=None)
-    assert 3 == topics.count()
+    cursor = await TopicService().search(lookup={})
+    topics = await cursor.to_list_raw()
+    assert 3 == len(topics)
 
-    folders, _ = app.data.find("user_topic_folders", req=None, lookup=None)
-    assert 2 == folders.count()
+    cursor = await UserFoldersResourceService().search(lookup={})
+    folders = await cursor.to_list_raw()
+    assert 2 == len(folders)
 
-    await client.delete(f"/users/{PUBLIC_USER_ID}")
+    # # TODO:- Test cases based on signal
```

Comment:
Can you use prefix `TODO-ASYNC`, that will allow us to know what tasks are left in the code by searching for this todo